soda_machine.rb

```diff
@@ -7,12 +7,21 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.count
   end
 
   def find_soda(soda_brand)
+    if @sodas
+    @sodas.index(soda_brand) do |i|
+      return @sodas[i]
+    end
```

Comment:
`index` either takes an item to match, or a block to tell them what criteria to use for deciding if you've found a match.  You shouldn't both pass it an argument AND give it a block.  Let's find some time to sit down and work through this challenge.